src/key/factory.js

```diff
@@ -266,61 +263,67 @@ async function wrapKeyObject(secretKeyPacket, secretSubkeyPackets, options) {
 }
 
 /**
- * Reads an unarmored OpenPGP key list and returns one or multiple key objects
+ * Reads an unarmored OpenPGP key and returns a key object
  * @param {Uint8Array} data to be parsed
- * @returns {Promise<{keys: Array<module:key.Key>,
- *            err: (Array<Error>|null)}>} result object with key and error arrays
+ * @returns {Promise<module:key.Key>} key object
  * @async
  * @static
  */
 export async function read(data) {
-  const result = {};
-  result.keys = [];
-  const err = [];
-  try {
-    const packetlist = new packet.List();
-    await packetlist.read(data);
-    const keyIndex = packetlist.indexOfTag(enums.packet.publicKey, enums.packet.secretKey);
-    if (keyIndex.length === 0) {
-      throw new Error('No key packet found');
-    }
-    for (let i = 0; i < keyIndex.length; i++) {
-      const oneKeyList = packetlist.slice(keyIndex[i], keyIndex[i + 1]);
-      try {
-        const newKey = new Key(oneKeyList);
-        result.keys.push(newKey);
-      } catch (e) {
-        err.push(e);
-      }
-    }
-  } catch (e) {
-    err.push(e);
-  }
-  if (err.length) {
-    result.err = err;
+  const packetlist = new packet.List();
+  await packetlist.read(data);
+  return new Key(packetlist);
```

Comment:
It would throw if the key is bad or armored. Open questions if we should catch it?

Reply:
I think it's better to throw than to return an error array. Most people don't check it, and are confused when the `keys` array is empty.